File: 프로그래머스/2/134239. 우박수열 정적분/우박수열 정적분.js
function solution(k, ranges) {
    let answer = [];
    let areas = [];
    
    while(k !== 1) {
        if(k % 2 === 0) {
            k /= 2;
            areas.push(k * 1.5);
        }
        else {
            let tmp = k;
            k = k * 3 + 1;
            areas.push(tmp + (k - tmp) * 0.5);
        }
    }
    
    for(const range of ranges) {
        let start = range[0];
        let finish = areas.length + range[1];
        
        if(start > finish) {
            answer.push(-1.0);
            continue;
        }
        if(start === finish) {
            answer.push(0.0);
            continue;
        }
        
        let sum = 0;
        for(let i = start; i < finish; i++) {
            sum += areas[i];
        }
        answer.push(sum);
    }
    
    return answer;
}

File: 프로그래머스/2/134239. 우박수열 정적분/우박수열 정적분.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<double> solution(int k, vector<vector<int>> ranges) {
    vector<double> answer;
    vector<double> areas;
    
    while(k != 1) {
        if(k % 2 == 0) {
            k /= 2;
            areas.push_back((double)(k * 1.5));
        }
        else {
            int tmp = k;
            k = k * 3 + 1;
            areas.push_back((double)tmp + ((double)(k - tmp) * 0.5));
        }
        
        //cout<<areas.back()<<' ';
    }
    
    //cout<<'\n';
    
    for(auto range: ranges) {
        int start = range[0];
        int finish = areas.size() + range[1];
        
        //cout<<start<<' '<<finish<<'\n';
        
        if(finish < start) {
            answer.push_back(-1.0);
            continue;
        }
        if(finish == start) {
            answer.push_back(0.0);
            continue;
        }
        
        double sum = 0;
        for(int i = start; i < finish; i++) {
            sum += areas[i];
        }
        
        answer.push_back(sum);
    }
    
    return answer;
}